### PR TITLE
Refactor button variants and classnames

### PR DESCRIPTION
**Pull request by avrilpearl**
## Description
Based on the updating of the global color variables and agreed upon button variants in our documentation, this pr refactors the button component classsnames - adding color variable modifiers to classnames and creating clear classname modifiers for primary, secondary, tertiary, and quaternary buttons based on usage.

Impact: Major Release.

## Related Issue / Ticket / Documentation

- Marked as WIP progress in documentation here: https://massds.gitbook.io/edit-mayflower-doc/components/atoms/buttons#classnames

## Steps to Test
* Key thing here is to confirm that this refactor will not negatively impact mass.gov @mrossi113 or @ygannett ?

Pattern Lab:
1. Button as Quaternary --> http://localhost:3000/?p=atoms-button-as-quaternary
2. Button as Tertiary --> http://localhost:3000/?p=atoms-button-as-tertiary
3. Button as Secondary --> http://localhost:3000/?p=atoms-button-as-secondary
4. Button as Secondary in Primary Alt Color --> http://localhost:3000/?p=atoms-button-as-secondary-primary-alt

React
1. npm install
2. npm start
3. Go to storybook and confirm below is possible.
button-refactor

#### Impacted Areas in Application
Anywhere using the button component, particularly the old button with outline which is refactored in this pr to secondary button.

### CODE REVIEW

**Review by mrossi113 — COMMENTED** [review]
I pulled this PR into Drupal. I'm seeing some random org page breaking. Need to look into this further 
```
The website encountered an unexpected error. Please try again later.

**Review by mrossi113 — APPROVED** [review]
👍 LGTM! After merging develop into the branch no 500 errors appeared on the Drupal side.

File: assets/scss/01-atoms/_buttons.scss [hunk 1]
```diff
@@ -14,40 +16,58 @@ button {
...
+  text-transform: none;
+  &:hover {
+   border-bottom: 3px solid $c-primary;
+   background-color: transparent;
+  }
 }
+
+.ma__button--disabled {
+  opacity: 0.5;
+  cursor: not-allowed;
```
Comment (clairesunstudio) [inline assets/scss/01-atoms/_buttons.scss:72]: disabled buttons should not allow hover state

File: assets/scss/01-atoms/_buttons.scss [hunk 2]
```diff
@@ -14,40 +16,58 @@ button {
...
 }
 
-.ma__button--minor {
-  @include ma-button("normal","outline",$c-primary);
+.ma__button--c-primary-alt {
+  @include ma-button("normal","solid",$c-primary-alt);
+}
+
+.ma__button--c-highlight {
+  @include ma-button("normal","solid",$c-highlight);
```
Comment (clairesunstudio) [inline assets/scss/01-atoms/_buttons.scss:28]: this highlight button is going to fail accessibility

File: assets/scss/01-atoms/_buttons.scss [hunk 3]
```diff
@@ -14,40 +16,58 @@ button {
...
 }
 
-.ma__button--minor.ma__button--secondary {
+.ma__button--secondary.ma__button--c-primary-alt {
   @include ma-button("normal","outline",$c-primary-alt);
 }
 
-.ma__button--tertiary {
-  @include ma-button("normal","solid",$c-highlight);
+.ma__button--secondary.ma__button--c-highlight {
```
Comment (clairesunstudio) [inline assets/scss/01-atoms/_buttons.scss:43]: this highlight button is going to fail accessibility

File: assets/scss/01-atoms/_buttons.scss (hunk 2 above)
Comment (clairesunstudio) [reply assets/scss/01-atoms/_buttons.scss:28]: for lighter colors the foreground color should use $c-black
screen shot 2018-09-19 at 4 40 43 pm

**Review by clairesunstudio — APPROVED** [review]
looks good, need to follow up on disabled state and highlight usage

**Pull request merged**